ReportingServicesTools/Functions/Admin/Set-RsDatabase.ps1

```diff
@@ -112,22 +149,45 @@ function Set-RsDatabase
             $username = $rsWmiObject.WindowsServiceIdentityActual
             $password = ''
         }
-        
+
         else
         {
-            if ($DatabaseCredential -eq $null)
+            if ($DatabaseCredential -eq $nul)
             {
                 throw "No Database Credential specified! Database credential must be specified when configuring $DatabaseCredentialType authentication."
             }
             $username = $DatabaseCredential.UserName
             $password = $DatabaseCredential.GetNetworkCredential().Password
         }
         #endregion Validating authentication and normalizing credentials
-        
+
+        #region Validating admin authentication and normalizing credentials
+        $adminUsername = ''
+        $adminPassword = $null
+        # default is service account
+        if ($AdminDatabaseCredentialType -like 'serviceaccount' -or -not $AdminDatabaseCredential)
```

Comment:
AdminCredentials should not be service account, the only valid credentials should be either current user for windows or SQL credentials (username and password)

Reply:
Makes sense. Fixing.